src/serializer.js

```diff
@@ -223,13 +179,33 @@ const MapHandler = [
   }),
 ];
 
-export const writer = (customWriters = [], opts = {}) => {
-  const ownHandlers = constructWriteHandlers(customWriters);
+export const writer = (appTypeWriters = [], opts = {}) => {
   const { verbose } = opts;
   const transitType = verbose ? 'json-verbose' : 'json';
 
   return transit.writer(transitType, {
-    handlers: transit.map([...ownHandlers, ...MapHandler]),
+    handlers: transit.map([...writeHandlers, ...MapHandler]),
+
+    // Use transform to transform app types to sdk types before sdk
+    // types are encoded by transit.
+    transform: v => {
+      if (v && v instanceof Object) {
+        if (v._sdkType) {
+          return toType(v);
+        }
+
+        const appTypeWriter = _.find(
+          appTypeWriters,
+          w => (w.appType && v instanceof w.appType) || (w.canHandle && w.canHandle(v))
```

Comment:
This one-liner is quite slow to grok. Could it be simplified a bit with ternaries etc?

Reply:
How? The equivalent version with ternaries would be:

```js
(w.appType ? v instanceof w.appType : false) || (w.canHandle ? w.canHandle(v) : false)
```

...which imo isn't any clearer.

Reply:
Yeah, keeping it fully as a one-liner is probably the problem. Your call, no need to change really.